src/blockdigest.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef _BITCOIN_SRC_BLOCKDIGEST_H_
+#define _BITCOIN_SRC_BLOCKDIGEST_H_
+
+#include <vector>
+#include <boost/filesystem.hpp>
+
+#include "bloom.h"
+#include "serialize.h"
+#include "uint256.h"
+
+class COutPoint;
+class CBlock;
+class CTransaction;
+
+namespace DigestFilter {
+
+enum format {
+    vectorized      = 1 << 0,   // argument is a uint8_t indicating size
+    mapped          = 1 << 1,   // argument comprises two uint8_t's indicating offset and count
+    inverted        = 1 << 2,   // listed elements are ABSENT; unlisted elements are PRESENT
+};
+
+typedef std::vector<uint8_t> data_t;
```

Comment:
I can't really use std::string as it's arbitrary byte data, not strings, right? Also, it's not always 32 bytes (e.g. for some p2sh script which could be > 32b or a 20 byte pub key hash).